src/cgutils.cpp

```diff
@@ -1486,20 +1338,20 @@ static Value *emit_getfield_knownidx(Value *strct, unsigned idx, jl_datatype_t *
             return typed_load(addr, ConstantInt::get(T_size, 0), jfty, ctx, tbaa, align);
         }
     }
-    else if (strct->getType()->isPointerTy()) { // something stack allocated
+    else if (strct.ispointer) { // something stack allocated
 #       ifdef LLVM37
         Value *addr = builder.CreateConstInBoundsGEP2_32(
-            cast<PointerType>(strct->getType()->getScalarType())->getElementType(),
+            cast<PointerType>(strct.V->getType()->getScalarType())->getElementType(),
             strct, 0, idx);
```

Comment:
k. only small fixes to get it to build on llvm trunk. it could probably use a rebase but i'll let you do that later ;-)